icd-main.cpp

```diff
@@ -28,6 +28,7 @@
 #include <sys/stat.h>
 #include <cstdio>
 #include <cstring>
+#include <unistd.h>
```

Comment:
Maybe I missed something, but it didn't compile for me. Complaining about `close` and `open` being undefined.